libsmtutil/CHCSmtLib2Interface.cpp

```diff
@@ -223,3 +225,289 @@ std::string CHCSmtLib2Interface::createHeaderAndDeclarations() {
+						SortPointer sort;
+						std::vector<smtutil::Expression> arguments;
+						if (isAtom(_subExpr.front()))
+						{
+							std::string const& op = asAtom(_subExpr.front());
+							if (op == "!")
+							{
+								// named term, we ignore the name
+								solAssert(_subExpr.size() > 2);
+								return toSMTUtilExpression(_subExpr[1]);
+							}
+							if (op == "exists" || op == "forall")
+							{
+								solAssert(_subExpr.size() == 3);
+								solAssert(!isAtom(_subExpr[1]));
+								return parseQuantifier(op, asSubExpressions(_subExpr[1]), _subExpr[2]);
+							}
+							for (size_t i = 1; i < _subExpr.size(); i++)
+								arguments.emplace_back(toSMTUtilExpression(_subExpr[i]));
+							if (auto tupleSort = lookupKnownTupleSort(op); tupleSort)
+							{
+								auto sortSort = std::make_shared<SortSort>(tupleSort.value());
+								return Expression::tuple_constructor(Expression(sortSort), arguments);
+							}
+							if (knownVariables.find(op) != knownVariables.end())
+							{
+								return smtutil::Expression(op, std::move(arguments), knownVariables.at(op));
+							}
+							else {
+								std::set<std::string> boolOperators{"and", "or", "not", "=", "<", ">", "<=", ">=",
+																	"=>"};
+								sort = contains(boolOperators, op) ? SortProvider::boolSort : arguments.back().sort;
+								return smtutil::Expression(op, std::move(arguments), std::move(sort));
+							}
+							smtAssert(false, "Unhandled case in expression conversion");
+						} else {
+							// check for const array
+							if (_subExpr.size() == 2 and !isAtom(_subExpr[0]))
+							{
+								auto const& typeArgs = asSubExpressions(_subExpr.front());
+								if (typeArgs.size() == 3 && typeArgs[0].toString() == "as" && typeArgs[1].toString() == "const")
+								{
+									auto arraySort = toSort(typeArgs[2]);
+									auto sortSort = std::make_shared<SortSort>(arraySort);
+									return smtutil::Expression::const_array(Expression(sortSort), toSMTUtilExpression(_subExpr[1]));
+								}
+								if (typeArgs.size() == 3 && typeArgs[0].toString() == "_" && typeArgs[1].toString() == "int2bv")
+								{
+									auto bvSort = std::dynamic_pointer_cast<BitVectorSort>(toSort(_subExpr[0]));
+									solAssert(bvSort);
+									return smtutil::Expression::int2bv(toSMTUtilExpression(_subExpr[1]), bvSort->size);
+								}
+								if (typeArgs.size() == 4 && typeArgs[0].toString() == "_")
+								{
+									if (typeArgs[1].toString() == "extract")
+									{
+										return smtutil::Expression(
+												"extract",
+												{toSMTUtilExpression(typeArgs[2]), toSMTUtilExpression(typeArgs[3])},
+												SortProvider::bitVectorSort // TODO: Compute bit size properly?
+												);
+									}
+								}
+							}
+
+							smtAssert(false, "Unhandled case in expression conversion");
+						}
+					}
+			}, _expr.data);
+		}
+	};
+
+
+#define precondition(CONDITION) if(!(CONDITION)) return {}
```

Comment:
What would be a better for to achieve this?
If I don't what to explicitly write 10 `if` checks like this
```
if (!<condition>)
    return {};
```

Yet, I want to check these conditions, but only bail out on further processing, not throwing an exception.
It feels like there currently isn't a better way than a macro, but style check does not like this.